hack/update-codegen.sh

```diff
@@ -18,42 +18,64 @@ set -o errexit
 set -o nounset
 set -o pipefail
 
-SCRIPT_ROOT=$(dirname "${BASH_SOURCE}")/..
+readonly SCRIPT_ROOT="$(cd "$(dirname "${BASH_SOURCE}")"/.. && pwd)"
 
-go install k8s.io/code-generator/cmd/{client-gen,lister-gen,informer-gen,deepcopy-gen,register-gen}
+readonly GO111MODULE="on"
+readonly GOFLAGS="-mod=vendor"
```

Comment:
Why do we need the vendor?

Reply:
It's about getting consistent behavior. We don't want this script to update `go.mod`, so I think that either `mod=vendor` or `mod=readonly` would probably be OK.

> By default, if the go version in go.mod is 1.14 or higher and a vendor directory is present, the go command will act as if -mod=vendor were used. Otherwise, the go command will act as if -mod=mod were used.

https://golang.org/ref/mod#build-commands

Reply:
`-mod=readonly` seems to work fine.